app/javascript/components/specifications-list/SpecsList.jsx

```diff
@@ -23,13 +22,26 @@ import { AppContext } from '../../contexts/AppContext';
 import { pageRoutes } from '../../services/pageRoutes';
 import { FILTER_OPTIONS, specsListStore } from './stores/specsListStore';
 import { i18n } from 'utils/i18n';
+import ExportMappings from './ExportMappings';
+import ConfigurationProfileSelect from '../shared/ConfigurationProfileSelect';
+import { useSelector } from 'react-redux';
+import { isMapper } from '../../helpers/Auth';
 
 const SpecsList = (_props) => {
-  const { currentConfigurationProfile, organization } = useContext(AppContext);
+  const { organization } = useContext(AppContext);
   const [state, actions] = useLocalStore(() => specsListStore());
-  const { mappings, filter, loading } = state;
+  const { configurationProfileId, mappings, filter, loading } = state;
 
-  useEffect(() => actions.fetchDataFromAPI(), [filter]);
+  const user = useSelector((state) => state.user);
+  const userIsMapper = isMapper(user);
+
+  useEffect(() => actions.fetchDataFromAPI(), [filter, configurationProfileId]);
+
+  useEffect(() => {
+    if (!userIsMapper) {
+      actions.setFilter('all');
```

Comment:
we can construct specsListStore with proper filter like `specsListStore(userIsMapper ? 'user' : 'all')` to avoid double querying. And if we still may have `configurationProfileId` change use `useDidMountEffect` here.

Reply:
Nice suggestion, but this change was reverted, can't implement it right now.